core/ArchiveProcessor/LoaderLock.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\ArchiveProcessor;
+
+use Piwik\Db;
+use Piwik\SettingsPiwik;
+
+class LoaderLock
+{
+
+    const MAX_LOCK_TIME = 60; //in seconds
+    protected $id;
+
+    public function __construct($id)
+    {
+        // for multi tenant database solution
+        $id = md5(SettingsPiwik::getPiwikInstanceId() . $id);
```

Comment:
It be great to hash the ID only if it's longer than 64 characters (too long). This will help better understand what lock is being set eg if we had to troubleshoot some locking issues directly in the database but also while debugging the code.

If the ID is longer than 64 characters, then we will want to do what I mentioned in https://github.com/matomo-org/matomo/pull/18243#discussion_r746152922

The benefit of something like `substr($id,0,32).md5($id)` is that most of the ID is still readable which is very useful when debugging.